Robust.Client/Graphics/Clyde/Clyde.HLR.cs

```diff
@@ -369,28 +354,39 @@ private void DrawEntities(Viewport viewport, Box2Rotated worldBounds, Box2 world
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private void ProcessSpriteEntities(MapId map, Matrix3 eyeMatrix, Box2Rotated worldBounds,
-            RefList<(SpriteComponent sprite, Matrix3 matrix, Angle worldRot, float yWorldPos)> list)
+        private void ProcessSpriteEntities(MapId map, Viewport view, IEye eye, Box2Rotated worldBounds,
+            RefList<(SpriteComponent sprite, Vector2 worldPos, Angle worldRot, Box2 spriteScreenBB)> list)
         {
+            // Construct a matrix equivalent for Viewport.WorldToLocal()
+
+            eye.GetViewMatrix(out var viewMatrix, view.RenderScale);
+
+            var uiProjmatrix = Matrix3.Identity;
+            uiProjmatrix.R0C0 = EyeManager.PixelsPerMeter;
+            uiProjmatrix.R1C1 = -EyeManager.PixelsPerMeter;
+            uiProjmatrix.R0C2 = view.Size.X / 2f;
+            uiProjmatrix.R1C2 = view.Size.Y / 2f;
+
+            var matrix = viewMatrix * uiProjmatrix;
+
             foreach (var comp in _entitySystemManager.GetEntitySystem<RenderingTreeSystem>().GetRenderTrees(map, worldBounds))
             {
                 var bounds = _entityManager.GetComponent<TransformComponent>(comp.Owner).InvWorldMatrix.TransformBox(worldBounds);
 
                 comp.SpriteTree.QueryAabb(ref list, (
-                    ref RefList<(SpriteComponent sprite, Matrix3 matrix, Angle worldRot, float yWorldPos)> state,
+                    ref RefList<(SpriteComponent sprite, Vector2 worldPos, Angle worldRot, Box2 spriteScreenBB)> state,
                     in SpriteComponent value) =>
                 {
                     var entity = value.Owner;
                     var transform = _entityManager.GetComponent<TransformComponent>(entity);
 
                     ref var entry = ref state.AllocAdd();
                     entry.sprite = value;
-                    entry.worldRot = transform.WorldRotation;
-                    entry.matrix = transform.WorldMatrix;
-                    var eyePos = eyeMatrix.Transform(new Vector2(entry.matrix.R0C2, entry.matrix.R1C2));
-                    // Didn't use the bounds from the query as that has to be re-calculated (and is probably more expensive than this).
-                    var bounds = value.CalculateBoundingBox(eyeMatrix);
-                    entry.yWorldPos = eyePos.Y - bounds.Extents.Y;
```

Comment:
Previously this got the sprite bounding box and just discarded it afterwards. Not it keeps it for use with the render target stuff, albeit in a different coordinate frame.

Reply:
I think this is going to break y-sorting, look at SpriteDrawingOrderComparer as I think it's using CompareTo on the bounds instead which I have no idea what would happen.

Reply:
Well it was `compare`-ing the `bottom` of the bounds, which should correspond to what it was calculating before (center.Y - extents.Y) But there was a bug as I had forgotten screen y-coordinates were flipped, so it should have been checking the top of the screen-bounds

Reply:
just checking with players & warped sprites, it seems to work fine atm